### PR TITLE
Yuhsuan/2023 compare release string

### LINKED ISSUE #2023: Unnecessary notification of the latest release
**Describe the bug**
We get unnecessary notification of the latest release when we compare the current frontend repo version v3.0.1 to the carta repo version v3.0.0. Maybe we need to ignore the patch version difference when we compare (assuming that we always update both repos when we have major/minor/beta updates).

**To Reproduce**
1. Run `app.preferenceStore.clearPreferences(["checkNewRelease", "latestRelease"]);` to reset the settings.
2. Reload frontend.
3. See the notification on the upper-right corner.

**Expected behavior**
No new notification.

**Screenshots or videos**
image

**Platform info (please complete the following information):**
 - OS [e.g. macOS Monterey]: macOS Big Sur
 - Browser [e.g. chrome, safari, electron app]: chrome
 - Browser version [e.g. 22]: 105
 - Backend branch [e.g. dev, v3b2 release]: dev
 - Frontend branch [e.g. dev, v3b2 release]: dev and v3.0.1

### PR DESCRIPTION
**Pull request by YuHsuan-Hwang**
**Description**

Closes #2023 by checking if responded version string is greater than the current version string stored in the preference.

Checked that all kinds of stable/beta version combinations work fine for `Semver.gt()`:
* stable vs stable
  `Semver.gt('v3.0.1', 'v3.0.1')` false
  `Semver.gt('v3.1.0', 'v3.0.1')` true
* beta vs stable
  `Semver.gt('v4.0.0-beta.1', 'v3.0.0')` true
  `Semver.gt('v4.0.0-beta.1', 'v3.0.1')` true
  `Semver.gt('v4.0.0', 'v4.0.0-beta.2')` true
  `Semver.gt('v4.1.0', 'v4.0.0-beta.2')` true
* beta vs beta
  `Semver.gt('v4.0.0-beta.2', 'v4.0.0-beta.1')` true
  `Semver.gt('v5.0.0-beta.1', 'v4.0.0-beta.1')` true
  `Semver.gt('v5.0.0-beta.1', 'v4.0.0-beta.2')` true

**Checklist**
- [ ] changelog updated / ~no changelog update needed~
- [ ] e2e test passing / ~added corresponding fix~
- [X] ~protobuf updated to the latest dev commit~ / no protobuf update needed

### CODE REVIEW

**Comment by kswang1029** [comment]
@ajm-asiaa is it possible to have `v4.0.0-beta1b`? In one of the v3beta we have such IIRC?

**Comment by ajm-ska** [comment]
@kswang1029 Yes, we have had such beta versions with a letter before. But if it would be hard to parse a letter in the version string, maybe we could try to avoid it in future? 
Perhaps we increment it even if it is only a small bug fix e.g. `v4.0.0-beta.1b` would be `v4.0.0-beta.2`? Or perhaps we could do something like`v4.0.0-beta1.1`?

**Comment by kswang1029** [comment]
> v4.0.0-beta1.1

this might work. @YuHsuan-Hwang @veggiesaurus @confluence suggestions?

**Comment by YuHsuan-Hwang** [comment]
> > v4.0.0-beta1.1
> 
> this might work. @YuHsuan-Hwang @veggiesaurus @confluence suggestions?

I think I can first check if `v4.0.0-beta.1b` works.

**Comment by YuHsuan-Hwang** [comment]
@kswang1029 @ajm-asiaa 
1. I revised from `v4.0.0-beta1` to `v4.0.0-beta.1` and retested the combinations. The new results are updated in the initial comment, all looking fine.
2. For `v4.0.0-beta.1b`, there's problem when we try to compare a number and a string:
`Semver.gt('v4.0.0-beta.2', 'v4.0.0-beta.1b')` false
3. If we change to `v4.0.0-beta.1.1`, so far the comparison looks fine:
`Semver.gt('v4.0.0-beta.1.1', 'v3.0.0')` true
`Semver.gt('v4.0.0-beta.1.1', 'v3.0.1')` true
`Semver.gt('v4.0.0', 'v4.0.0-beta.2.1')` true
`Semver.gt('v4.1.0', 'v4.0.0-beta.2.1')` true
`Semver.gt('v4.0.0-beta.1', 'v4.0.0-beta.1.1')` false
`Semver.gt('v4.0.0-beta.2', 'v4.0.0-beta.1.1')` true
`Semver.gt('v4.0.0-beta.2.1', 'v4.0.0-beta.1.1')` true
`Semver.gt('v4.0.0-beta.1.2', 'v4.0.0-beta.1.1')` true

File: src/stores/AppStore.ts [hunk 1]
```diff
@@ -2,6 +2,7 @@ import * as _ from "lodash";
 import {action, autorun, computed, flow, makeObservable, observable, ObservableMap, when} from "mobx";
 import * as Long from "long";
 import axios from "axios";
+import * as Semver from "semver";
```
Comment (veggiesaurus) [inline src/stores/AppStore.ts:5]: shouldn't there be an additional `package.json` change too?

Comment (YuHsuan-Hwang) [reply src/stores/AppStore.ts:5]: @veggiesaurus added `semver` to `package.json`. also updated `package-lock.json`

**Review by ajm-ska — APPROVED** [review]
I also noticed the issue that this PR is addressing on our carta-controller deployments (cartaviewer and the e-disk server). It reports that the Current release is 3.0.1 and the New release is 3.0.0.

I think the modifications in this PR should work fine.

**Review by kswang1029 — APPROVED** [review]

**Comment by kswang1029** [comment]
> 2\. For `v4.0.0-beta.1b`, there's problem when we try to compare a number and a string:
>     `Semver.gt('v4.0.0-beta.2', 'v4.0.0-beta.1b')` false

Let's try to avoid this naming rule then.
